src/azure-account.ts

```diff
@@ -25,6 +25,7 @@ import TelemetryReporter from 'vscode-extension-telemetry';
 const localize = nls.loadMessageBundle();
 
 const keytar = getNodeModule<typeof keytarType>('keytar');
+var selectedEnvironment : AzureEnvironment;
```

Comment:
Instead of using a global variable, there should usually be an AzureSession object in any context where an environment is needed. Please use that. (That will keep the code more flexible towards multiple concurrent logins.)

Reply:
With that you can then move reading and storing of the environment in keytar to the same places where the refresh token is read and stored.